### PR TITLE
Navigation to diff view from description page

### PR DESCRIPTION
**Pull request by rebornix**
nagivation-diff-view-from-description-page

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -223,10 +223,10 @@ export function registerCommands(context: vscode.ExtensionContext, prManager: IP
 		return await prManager.requestChanges(pr, message);
 	}));
 
-	context.subscriptions.push(vscode.commands.registerCommand('pr.openDescription', async (pr: IPullRequestModel) => {
-		const pullRequest = ensurePR(prManager, pr);
+	context.subscriptions.push(vscode.commands.registerCommand('pr.openDescription', async (descriptionNode: DescriptionNode) => {
+		const pullRequest = ensurePR(prManager, descriptionNode.pullRequestModel);
```
Comment (RMacfarlane) [inline src/commands.ts]: when opening the description from the command palette or the status bar, descriptionNode can be undefined

Comment (rebornix) [reply src/commands.ts]: fixed ;)

**Comment by rebornix** [comment]
Added outdated label to comment reviews and we don't support code navigation if the comment is outdated.

image

image

File: preview-src/pullRequestOverviewRenderer.ts [hunk 2]
```diff
@@ -591,11 +591,24 @@ class ReviewNode {
...
+
 					const diffView: HTMLDivElement = document.createElement('div');
 					diffView.className = 'diff';
 					const diffHeader: HTMLDivElement = document.createElement('div');
 					diffHeader.className = 'diffHeader';
-					diffHeader.textContent = comments[0].path;
+					const diffPath: HTMLSpanElement = document.createElement('span');
+					diffPath.className =  outdated ? 'diffPath outdated' : 'diffPath';
+					diffPath.textContent = comments[0].path;
+					diffPath.addEventListener('click', () => this.openDiff(comments[0]));
```
Comment (RMacfarlane) [inline preview-src/pullRequestOverviewRenderer.ts]: should the event listener only be added if the comment isn't outdated?

File: src/commands.ts (hunk 1 above)
Comment (RMacfarlane) [reply src/commands.ts]: nice

File: src/github/pullRequestOverview.ts [hunk 3]
```diff
@@ -268,6 +275,18 @@ export class PullRequestOverviewPanel {
...
 
+	private openDiff(message: IRequestMessage<{ comment: Comment }>): void {
+		try {
+			const comment = message.args.comment;
+			const prContainer = this._descriptionNode.parent;
+
+			if ((prContainer as TreeNode | Revealable<TreeNode>).revealComment) {
+				(prContainer as TreeNode | Revealable<TreeNode>).revealComment(comment);
+			}
+		} catch (e) {
```
Comment (RMacfarlane) [inline src/github/pullRequestOverview.ts]: log something to debug if reveal fails?

File: preview-src/pullRequestOverviewRenderer.ts [hunk 4]
```diff
@@ -591,11 +591,24 @@ class ReviewNode {
 						});
 					}
 
+					let outdated = comments[0].position !== comments[0].original_position;
```
Comment (RMacfarlane) [inline preview-src/pullRequestOverviewRenderer.ts]: in other places we check if position is null to determine if a comment is outdated. I think we should do that here too

Comment (rebornix) [reply preview-src/pullRequestOverviewRenderer.ts]: done

File: preview-src/pullRequestOverviewRenderer.ts (hunk 2 above)
Comment (rebornix) [reply preview-src/pullRequestOverviewRenderer.ts]: done

**Pull request merged**